### ISSUE TITLE
Expose promiseRequest in API

### ISSUE THREAD

**Issue by joelrich** [jaggedsoft/node-binance-api#424]
I have been a couple of times in the situation that I wanted to add/change a futures API. E.g. to react to changes in the Binance REST API more quickly or to add an API which is not yet supported. For this, I forked the repo and exported promiseRequest and fapi. Would you accept a pull request with these changes? Then I don't have to maintain my fork anymore.

**Comment by jaggedsoft** [comment]
thanks for the great ideas, I will publish asap. ciao!

**Comment by joelrich** [comment]
Cool, thank you! Can you also expose the following constants?
```js
    let base = 'https://api.binance.com/api/';
    let wapi = 'https://api.binance.com/wapi/';
    let sapi = 'https://api.binance.com/sapi/';
    let fapi = 'https://fapi.binance.com/fapi/';
```
The you can do something like:
```js
            let transferToSpot = await this.binance.promiseRequest("v1/futures/transfer",
                {asset: "USDT", amount: missingInUsdt, type: 2},
                {base: this.binance.sapi, type:'SIGNED', method:'POST'});
```
(This is a real example from my code base)

**Comment by jaggedsoft** [comment]
Yeah good idea. maybe we could have `binance.domain` return an object with these properties available for read or write: `base, wapi, sapi, fapi, fapiTest, stream, combineStream, fstream, fstreamSingle` (all currently available endpoint uri's)

**Comment by joelrich** [comment]
Such an object sounds like a good idea. I would call it `binance.endpoints` though.
